src/signature.js

```diff
@@ -71,36 +70,53 @@ angular.module('signature').directive('signaturePad', ['$window', '$timeout',
         }
       ],
       link: function (scope, element, attrs) {
-        canvas = element.find('canvas')[0];
-        scope.signaturePad = new SignaturePad(canvas);
+        var canvas = element.find('canvas')[0];
+        var parent = canvas.parentElement;
+        var ctx = canvas.getContext('2d');
+
+        var width = parseInt(scope.width, 10);
+        var height = parseInt(scope.height, 10);
+        var aspectRatio = height / width;
+
+        canvas.width = width;
+        canvas.height = height;
         
         //Resize to fix UI Bootstrap Modal Show problem
         $timeout(function(){
```

Comment:
In the beginning i was thinking to do that, but this issue about resizing. Maybe we should do optimizations in separate pull request?